Source/SwiftyDropbox/Platform/SwiftyDropbox_iOS/OAuthMobile.swift

```diff
@@ -146,6 +196,24 @@ open class DropboxMobileOAuthManager: DropboxOAuthManager {
         }
         return false
     }
+
+    /// Creates a string that contains all code flow query parameters.
+    private static func createExtraQueryParamsString(for authSession: AuthSession) -> String {
+        let pkceData = authSession.pkceData
+        var extraQueryParams = "\(OAuthConstants.codeChallengeKey)=\(pkceData.codeChallenge)"
+            + "&\(OAuthConstants.codeChallengeMethodKey)=\(pkceData.codeChallengeMethod)"
+            + "&\(OAuthConstants.tokenAccessTypeKey)=\(authSession.tokenAccessType)"
+            + "&\(OAuthConstants.responseTypeKey)=\(authSession.responseType)"
+        if let scopeRequest = authSession.scopeRequest {
+            if let scopeString = scopeRequest.scopeString {
+                extraQueryParams += "&\(OAuthConstants.scopeKey)=\(scopeString)"
+            }
+            if scopeRequest.includeGrantedScopes {
+                extraQueryParams += "&\(OAuthConstants.includeGrantedScopesKey)=\(scopeRequest.scopeType.rawValue)"
+            }
+        }
+        return extraQueryParams
```

Comment:
this whole file is using URLComponents which is the proper way to mutate URL imo, but then this 1 method outputs a string which you use to append more queryItems... this method directly encodes the queryParams into 1 `URLQueryItem` in `dAuthURL` as 1 single string which is the opposite of how URLQueryItem is intended.

Why not just modify `dAuthURL` to correctly add these 6 query parameters as seperate URLQueryItems.

Reply:
Good question, this is an unusually use case. The way delgated auth (dauth) works is to use DBApp as a relay to send all the params from SDK to the server. If we send the params separately, an update in the SDK would require an update in DBApp too, this would lead to a situation where a 3rd party app with new version SDK fails to auth with old DBApp, e.g. old DBApp don't understand new param hence not sending to server. Packing all extra params into one makes it so that changes in SDK don't require changes in DBApp.